Updating the resource reference schemas for environment and secret store (#1076)
radius-project/docs: docs/content/reference/resource-schema/core-schema/environment-schema/index.md

| kind | y | The kind of identity. 'azure.com.workload' is currently only supported. | `'azure.com.workload'` |
| oidcIssuer | n | The [OIDC issuer URL](https://azure.github.io/azure-workload-identity/docs/installation/self-managed-clusters/oidc-issuer.html) for your Kubernetes cluster. | `'{IssuerURL}/.well-known/openid-configuration'` |

### recipeConfig

| Key  | Required | Description | Example |
|------|:--------:|-------------|---------|
| terraform | y | Configuration for Terraform Recipes. Controls how Terraform plans and applies templates as part of Recipe deployment. | [See below](#terraform-properties)

> | env | n | Environment variables injected during Terraform Recipe execution for the recipes in the environment. | [TODO](#TODO)

Author: Need help with what an example would be here

Reviewer: Draft:
### env Properties

| Key  | Required | Description | Example |
|------|:--------:|-------------|---------|
| \<user-defined key-value pairs\> | n | User-defined environment variables. | `'env_var_1'`: `'env_value_1'`

#### terraform properties

| Key  | Required | Description | Example |
|------|:--------:|-------------|---------|
| authentication | y | Authentication information used to access private Terraform module sources. Supported module sources: Git. | [See below](#authentication-properties)

> | providers | n | Configuration for Terraform Recipe Providers. Controls how Terraform interacts with cloud providers, SaaS providers, and other APIs. | For more information, please see: [https://developer.hashicorp.com/terraform/language/providers/configuration](https://developer.hashicorp.com/terraform/language/providers/configuration).

Reviewer 2: ```suggestion
| providers | n | Configuration for Terraform Recipe Providers. Controls how Terraform interacts with cloud providers, SaaS providers, and other APIs. | For more information refer to the [Terraform documentation](https://developer.hashicorp.com/terraform/language/providers/configuration).
```

##### authentication properties

| Key  | Required | Description | Example |
|------|:--------:|-------------|---------|
| pat | y | Personal Access Token (PAT) configuration used to authenticate to Git platforms. | [See below](#pat-properties)

##### pat properties

| Key  | Required | Description | Example |
|------|:--------:|-------------|---------|

> | secret | y | The ID of an Applications.Core/SecretStore resource containing the Git platform personal access token (PAT). The secret store must have a secret named 'pat', containing the PAT value. A secret named 'username' is optional, containing the username associated with the pat. By default no username is specified. | For more information, please see: [https://developer.hashicorp.com/terraform/language/providers/configuration](https://developer.hashicorp.com/terraform/language/providers/configuration).

Reviewer 2: ```suggestion
| secret | y | The ID of an Applications.Core/SecretStore resource containing the Git platform personal access token (PAT). The secret store must have a secret named 'pat', containing the PAT value. A secret named 'username' is optional, containing the username associated with the pat. By default no username is specified. | For more information refer to the [Terraform documentation](https://developer.hashicorp.com/terraform/language/providers/configuration).
```
